proposals/dao/repay_fuse_bad_debt.ts

```diff
@@ -0,0 +1,172 @@
+// IE check balances, check state of contracts, etc.
+const validate: ValidateUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  const debtor = '0x32075bAd9050d4767018084F0Cb87b3182D36C45';
+
+  const ctokens = [
+    '0xd8553552f8868C1Ef160eEdf031cF0BCf9686945', // Pool 8: FEI
+    '0xbB025D470162CC5eA24daF7d4566064EE7f5F111', // Pool 8: ETH
+    '0x7e9cE3CAa9910cc048590801e64174957Ed41d43', // Pool 8: DAI
+    '0x7259eE19D6B5e755e7c65cECfd2466C09E251185', // Pool 8: wstETH
+    '0x647A36d421183a0a9Fa62717a64B664a24E469C7', // Pool 8: LUSD
+    '0xFA1057d02A0C1a4885851e3F4fD496Ee7D38F56e', // Pool 18: ETH
+    '0x8E4E0257A4759559B4B1AC087fe8d80c63f20D19', // Pool 18: DAI
+    '0x6f95d4d251053483f41c8718C30F4F3C404A8cf2', // Pool 18: USDC
+    '0x3E5C122Ffa75A9Fe16ec0c69F7E9149203EA1A5d', // Pool 18: FRAX
+    '0x17b1A2E012cC4C31f83B90FF11d3942857664efc', // Pool 18: FEI
+    '0x51fF03410a0dA915082Af444274C381bD1b4cDB1', // Pool 18: RAI
+    '0xB7FE5f277058b3f9eABf6e0655991f10924BFA54', // Pool 18: USTw
+    '0x9de558FCE4F289b305E38ABe2169b75C626c114e', // Pool 27: FRAX
+    '0xda396c927e3e6BEf77A98f372CE431b49EdEc43D', // Pool 27: FEI
+    '0xF148cDEc066b94410d403aC5fe1bb17EC75c5851', // Pool 27: ETH
+    '0x0C402F06C11c6e6A6616C98868A855448d4CfE65', // Pool 27: USTw
+    '0x26267e41CeCa7C8E0f143554Af707336f27Fa051', // Pool 127: ETH
+    '0xEbE0d1cb6A0b8569929e062d67bfbC07608f0A47', // Pool 127: USDC
+    '0x4B68ef5AB32261082DF1A6C9C6a89FFD5eF168B1', // Pool 127: DAI
+    '0xe097783483D1b7527152eF8B150B99B9B2700c8d', // Pool 127: USDT
+    '0x0F0d710911FB37038b3AD88FC43DDAd4Edbe16A5', // Pool 127: USTw
+    '0x8922C1147E141C055fdDfc0ED5a119f3378c8ef8', // Pool 127: FRAX
+    '0x7DBC3aF9251756561Ce755fcC11c754184Af71F7', // Pool 144: ETH
+    '0x3a2804ec0Ff521374aF654D8D0daA1d1aE1ee900', // Pool 144: FEI
+    '0xA54c548d11792b3d26aD74F5f899e12CDfD64Fd6', // Pool 144: FRAX
+    '0xA6C25548dF506d84Afd237225B5B34F2Feb1aa07', // Pool 144: DAI
+    '0xfbD8Aaf46Ab3C2732FA930e5B343cd67cEA5054C', // Pool 146: ETH
+    '0x49dA42a1EcA4AC6cA0C6943d9E5dc64e4641e0E3', // Pool 146: wstETH
+    '0xe14c2e156A3f310d41240Ce8760eB3cb8a0dDBE3', // Pool 156: USTw
+    '0x001E407f497e024B9fb1CB93ef841F43D645CA4F', // Pool 156: FEI
+    '0x5CaDc2a04921213DE60B237688776e0F1A7155E6', // Pool 156: FRAX
+    '0x9CD060A4855290bf0c5aeD266aBe119FF3b01966', // Pool 156: DAI
+    '0x74897C0061ADeec84D292e8900c7BDD00b3388e4', // Pool 156: LUSD
+    '0x88d3557eB6280CC084cA36e425d6BC52d0A04429', // Pool 156: USDC
+    '0xe92a3db67e4b6AC86114149F522644b34264f858' // Pool 156: ETH
+  ];
+
+  for (const ctokenAddress of ctokens) {
+    const ctoken = await hre.ethers.getContractAt('CTokenFuse', ctokenAddress);
+    const debt = await ctoken.callStatic.borrowBalanceCurrent(debtor);
+
+    if (debt.gt(0)) {
+      throw new Error(`Debt for ctoken ${ctokenAddress} is greater than 0: ${debt}`);
+    }
+  }
```

Comment:
I would probably make that `fuseFixer.getTotalDebt()` function on the contract public and assert it returns 0 here too

Reply:
Maybe have it emit an event with the total debt and catch that event + inspect the value

Reply:
Great call